NLP/main/extract_display_quotes.py

```diff
@@ -0,0 +1,531 @@
+        '''
+        Initiate the QuotationTool
+        '''
+        # initiate the app_logger
+        self.app_logger = utils.create_logger('quote_extractor', log_dir='logs', 
+                                         logger_level=logging.INFO,
+                                         file_log_level=logging.INFO)
+        
+        # download spaCy's en_core_web_lg, the pre-trained English language tool from spaCy
+        print('Loading spaCy language model...')
+        self.nlp = spacy.load('en_core_web_lg')
+        print('Finished loading.')
+        
+    
+    def upload_files(self, file_type):
+        '''
+        Upload text or excel files as input to the QuotationTool
+        
+        Args:
+            file_type: the file type being uploaded ('text' for uploading .txt files
+                                                     or 'excel' for uploading .xlsx file)
+        '''
+        if file_type=='text':
+            print('Upload your .txt files here:')
+            # widget to upload .txt files
+            uploader = widgets.FileUpload(
+                accept='.txt', # accepted file extension 
+                multiple=True  # True to accept multiple files
+                )
+        elif file_type=='excel':
+            print('Upload your excel spreadsheet (.xlsx) here:')
+            # widget to upload .xlsx file
+            uploader = widgets.FileUpload(
+                accept='.xlsx', # accepted file extension
+                multiple=False  # to accept one Excel file only
+                )
+        else:
+            # warning if other type files are uploaded
+            print('You can only upload .txt or .xlsx files!')
+        
+        # give notification when file is uploaded
+        def _cb(change):
+            clear_output()
+            print('File uploaded!')
+            
+        uploader.observe(_cb, names='data')
+        
+        return uploader
+    
+    
+    def nlp_preprocess(self, text):
+        '''
+        Pre-process and create spaCy text
+
+        Args:
+            text: the text to be processed
+        '''
+        # pre-process text
+        text = sent_tokenize(text)
+        text = ' '.join(text)
+        text = utils.preprocess_text(text)
+        
+        # apply the spaCy's tool to the text
+        doc = self.nlp(text)
+        
+        return doc
+    
+    
+    def process_txt(self, txt_upload):    
+        '''
+        Pre-process uploaded .txt files into pandas dataframe
+
+        Args:
+            txt_upload: the uploaded .txt files from upload_files()
+        '''
+        # create an empty list for a placeholder to store all the texts
+        all_files = []
+        
+        # search for text files (.txt) inside the folder and extract all the texts
+        for input_file in txt_upload.value.keys():
+            text_dict = {}
+            
+            # use the text file name as the doc_id
+            doc_id = input_file.replace('.txt', '')
```

Comment:
Need to consider sometimes filenames can be in different cases, e.g. all in upper case .TXT.
Lower() the input_file before replacing '.txt'.
Though still think it's worth to generate doc_id rather than taking the filename. The filename can be extra long, or duplicated if sourced from different folders.

Reply:
lower() added to code. also refer to previous comments